registration-processor/core-processor/registration-processor-abis-handler-stage/src/test/java/io/mosip/registration/processor/abis/handler/stage/test/AbisHandlerStageTest.java

```diff
@@ -1,500 +1,738 @@
+
+//		Mockito.when(packetManagerService.getBiometrics(any(),any(),any(), any(),any())).thenReturn(null);
+		mockDataSharePolicy(Lists.newArrayList(BiometricType.FACE, BiometricType.IRIS, BiometricType.FINGER));
+
+		setMetaInfoMap(Arrays.asList("leftEye"));
+//		setMetaInfoMap(Arrays.asList("rightMiddle"));
+		MessageDTO dto = new MessageDTO();
+		dto.setRid("10003100030001520190422074511");
+		MessageDTO result = abisHandlerStage.process(dto);
+
+		assertFalse(result.getIsValid());
+	}
+	
+	@Test
+	public void biometricsTypeNotFoundConfig()
+			throws ApisResourceAccessException, PacketManagerException, JsonProcessingException, IOException {
+
+		defaultMockToProcess();
+
+//		Mockito.when(packetManagerService.getBiometrics(any(),any(),any(), any(),any())).thenReturn(null);
+		mockDataSharePolicy(Lists.newArrayList(BiometricType.FOOT));
+
+		setMetaInfoMap(Arrays.asList("leftEye"));
+//		setMetaInfoMap(Arrays.asList("rightMiddle"));
+		MessageDTO dto = new MessageDTO();
+		dto.setRid("10003100030001520190422074511");
+		MessageDTO result = abisHandlerStage.process(dto);
+
+		assertFalse(result.getIsValid());
+	}
+
+
+	@Test
+	public void emptyBdbFound()
+			throws ApisResourceAccessException, PacketManagerException, JsonProcessingException, IOException {
+
+		defaultMockToProcess();
+
+		List<BIR> birTypeList = new ArrayList<>();
+		BIR birType1 = new BIR.BIRBuilder().build();
+		BDBInfo bdbInfoType1 = new BDBInfo.BDBInfoBuilder().build();
+		io.mosip.kernel.biometrics.entities.RegistryIDType registryIDType = new RegistryIDType();
+		registryIDType.setOrganization("Mosip");
+		registryIDType.setType("257");
+		io.mosip.kernel.biometrics.constant.QualityType quality = new QualityType();
+		quality.setAlgorithm(registryIDType);
+		quality.setScore(90l);
+		bdbInfoType1.setQuality(quality);
+		BiometricType singleType1 = BiometricType.FINGER;
+		List<BiometricType> singleTypeList1 = new ArrayList<>();
+		singleTypeList1.add(singleType1);
+		List<String> subtype1 = new ArrayList<>(Arrays.asList("Left", "RingFinger"));
+		bdbInfoType1.setSubtype(subtype1);
+		bdbInfoType1.setType(singleTypeList1);
+		birType1.setBdbInfo(bdbInfoType1);
+		Map<String, Object> others = new HashMap<>();
+		others.put("EXCEPTION", false);
+		birType1.setBdb(null);
+		birTypeList.add(birType1);
+
+		BiometricRecord biometricRecord = new BiometricRecord();
+		biometricRecord.setSegments(birTypeList);
+
+		Mockito.when(packetManagerService.getBiometrics(any(), any(), any(), any(), any())).thenReturn(biometricRecord);
+		mockDataSharePolicy(Lists.newArrayList(BiometricType.IRIS, BiometricType.FINGER, BiometricType.FACE));
+
+		MessageDTO dto = new MessageDTO();
+		dto.setRid("10003100030001520190422074511");
+		MessageDTO result = abisHandlerStage.process(dto);
+
+		assertFalse(result.getIsValid());
+	}
+
+	private void defaultMockToProcess() {
+		Mockito.when(registrationStatusService.getRegistrationStatus(any())).thenReturn(registrationStatusDto);

```

Comment:
Can't we add all these in setup ?

Reply:
No this default mock was specific for 4 test cases